pkg/types/validation/installconfig_test.go

```diff
@@ -1095,6 +1096,18 @@ func TestValidateInstallConfig(t *testing.T) {
 			}(),
 			expectedError: `^credentialsMode: Unsupported value: "Mint": supported values: "Manual"$`,
 		},
+		{
+			name: "invalid azurestack platform RHCOSImage field",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Platform = types.Platform{
+					Azure: validAzureStackPlatform(),
+				}
+				c.Azure.RHCOSImage = "invalid-image"
+				return c
+			}(),
+			expectedError: `RHCOSImage must be in the ASH environment`,
```

Comment:
use the regex pattern of [^] [message] [$] to ensure this is the complete error message.